common/djangoapps/course_modes/admin.py

```diff
@@ -29,12 +29,15 @@
 from lms.djangoapps.verify_student import models as verification_models
 
 
-class CourseModeForm(forms.ModelForm):
-
-    class Meta(object):
-        model = CourseMode
-        fields = '__all__'
+COURSE_MODE_SLUG_CHOICES = []
+try:
+    COURSE_MODE_SLUG_CHOICES = [(mode.mode_slug, mode.mode_slug) for mode in EnrollmentMode.objects.all()]
+except:  # pylint: disable=bare-except
+    pass
 
+if not COURSE_MODE_SLUG_CHOICES:
+    # The entries in this table are populated in a migration file, and migration files do not
+    # run as part of unit tests. Therefore, for tests hard-code the choices.
```

Comment:
I was chatting with @ormsbee, and there are general problems with me trying to access a database at the time that the file is loaded (with future versions of django). Plus edx-platform would have to be restarted for any new EnrollmentMode instances to be recognized (both here and in verified_track_content/partition_scheme.py).

I could try to make both of these usages lazy, but then I still have to figure out how to populate them in all unit tests which need them (a much bigger deal for verified_track_content/partition_scheme.py than for the admin table!).

@bderusha sorry, but I think I am going to abandon the database table approach and go with a setting that has the slug/ID mapping. :(